main/src/library/RedBlackTree.flix

```diff
@@ -643,4 +643,105 @@ namespace RedBlackTree {
             let b1 = seqCount(f, b);
             a1 + v1 + b1
     }
+
+    ///
+    /// Optionally finds `k -> v` where `k` is the smallest key according to the given comparator `cmp`.
+    ///
+    /// Returns `None` if `m` is empty.
+    ///
+    /// Purity polymorphic: Runs in parallel when given a pure function `f`.
+    ///
+    @ParallelWhenPure
+    pub def minimumBy(cmp: (k, v, k, v) -> Comparison & ef, t: RedBlackTree[k, v]): Option[(k, v)] & ef =
```

Comment:
Second question: Does this belong here (i.e. the use of reifyEff) or should we move it to Set?

I guess an argument for keeping it is that we can reuse it in both Set and Map.

Reply:
Hmm, what was your original thought / reasoning for putting delegating this function to RBTree in #2054?

Reply:
The parallel versions definitely needs to be in RBTree. Agree? The question is about the other stuff.

Reply:
Ahh, so you would want to call `reifyEff` in `Map` / `Set` and then delegate to the correct function? That could make sense, but should some future data structure depend on this function it would also have the benefit of being parallel. However, that's not really a solid argument.
I'm honestly not sure what to do. Maybe we should decide later? Your call.

Reply:
Yes. I was thinking of moving reify in Map/Set and then delegate from there.

Reply:
Actually, let us do that. We keep parWhatever and parSeqWhatevr in RBTree. Then we reifyEff inside Set/Map. That way we push the "magic" into Set/Map.

Reply:
Roger roger. Will fix that tomorrow :)

Reply:
Actually, I suggest we merge this and then I'll make a PR that fixes this for both `map`, `count` and `minimumBy`